saleor/core/tasks.py

```diff
@@ -17,13 +17,15 @@ def delete_from_storage_task(path):
 @app.task
 def delete_event_payloads_task():
     event_payload_delete_period = timezone.now() - EVENT_PAYLOAD_DELETE_PERIOD
-    EventDeliveryAttempt.objects.filter(
-        created_at__lte=event_payload_delete_period
-    ).delete()
-    EventDelivery.objects.filter(created_at__lte=event_payload_delete_period).delete()
-    EventPayload.objects.filter(
-        deliveries__isnull=True, created_at__lte=event_payload_delete_period
-    ).delete()
+    time_filter = {"created_at__lte": event_payload_delete_period}
+    attempts_queryset = EventDeliveryAttempt.objects.filter(**time_filter)
+    deliveries_queryset = EventDelivery.objects.filter(**time_filter)
+    payloads_queryset = EventPayload.objects.filter(
+        deliveries__isnull=True, **time_filter
+    )
```

Comment:
Are you sure this is going to work? we later delete the deliveries using raw method and after that it will evaluate this qs for deletion, not sure how it will behave since first _raw_delete (on deliveries) don't mind cascades?